src/mrsimulator/utils/cartesian_tensor.py

```diff
@@ -0,0 +1,363 @@
+    # Calculate the eigenvalues and eigenvectors of the traceless tensor
+    eigenvalues, L = np.linalg.eigh(traceless_tensor)
+    idx = np.argsort(
+        np.abs(eigenvalues)
+    )  # Sort the eigenvalues by their absolute values
+    idx[0], idx[1] = idx[1], idx[0]  # Swap the first two indices
+    eigenvalues = eigenvalues[idx]  # Swap the eigenvalues
+    L = L[:, idx]  # Swap the columns of the eigenvector matrix
+    # Flip the sign of the eigenvectors if the determinant is negative
+    if np.linalg.det(L) < 0:
+        L = -L
+
+    r = R.from_matrix(L)
+    euler_angles = r.as_euler("zyz")
+
+    zeta_sigma = eigenvalues[2]
+    eta_sigma = (eigenvalues[1] - eigenvalues[0]) / eigenvalues[2]
+
+    return euler_angles, zeta_sigma, eta_sigma, isotropic_component
+
+
+def from_haeberlen_params(
+    euler_angles: List[float],
+    zeta_sigma: float,
+    eta_sigma: float,
+    isotropic_component: float,
+) -> np.ndarray:
+    """Haeberlen parameters to a 3x3 symmetric Cartesian tensor.
+
+    Args:
+        euler_angles:
+            An ndarray of three Euler angles [``alpha``, ``beta``, ``gamma``]
+            using the "zyz" convention.
+        zeta_sigma:
+            Anisotropy parameter.
+        eta_sigma:
+            Asymmetry parameter
+        isotropic_component:
+            Isotropic parameter
+
+    Returns:
+        tensor: A 3x3 np.ndarray of symmetric Cartesian tensor.
+    """
+    eigenvalues = np.array(
+        [
+            -zeta_sigma * (1 + eta_sigma) / 2,
+            -zeta_sigma * (1 - eta_sigma) / 2,
+            zeta_sigma,
+        ]
+    )
+
+    # Assuming alpha_s, beta_s, gamma_s are your Euler angles
+    r = R.from_euler("zyz", euler_angles)
+    L = r.as_matrix()
+    diag_matrix = np.diag(eigenvalues)
+
+    tensor = L @ diag_matrix @ L.T
+
+    tensor += isotropic_component * np.eye(3)
+    return tensor
+
+
+def mehring_principal_components_to_maryland(
+    lambdas: List[float],
+) -> Tuple[float, float, float]:
+    """Mehring principal components to  Maryland.
+
+    Args:
+        lambdas:
+            Eigenvalues from Mehring convention.
+
+    Returns:
+        A tuple of (span, skew, isotropic) in Maryland convention where
+        ``span`` is the width, ``skew`` is the asymmetry, and ``isotropic``
+        is the isotropic components.
+    """
+    isotropic = np.mean(lambdas)
+    span = lambdas[2] - lambdas[0]
+    skew = 3 * (isotropic - lambdas[1]) / span
+    return span, skew, isotropic
+
+
+def maryland_to_mehring_principal_components(
+    isotropic: float, span: float, skew: float
+) -> np.ndarray:
+    """Maryland components to Mehring principal components.
+
+    Args:
+        span:
+            Maryland anisotropy component.
+        skew:
+            Maryland asymmetry component.
+        isotropic:
+            Maryland isotropic component.
+
+    Returns:
+        A ndarray of Mehring Eigenvalues.
+    """
+    lambda1 = isotropic - span
```

Comment:
I was writing the test for maryland_to_mehring and vice-versa and couldn't get it to work. I think  there is  an issue with `lambda1` equation. 

Revesing `mehring_principal_components_to_maryland` convertion, lambda1 should  be

```py
lambda1 = (-span + 2 * isotropic + skew * span / 3) / 2
```

Can you confirm?

Reply:
yes, the lambda1 expression is wrong.  It should be lambda1 = lambda3 - span, so it should be calculated after lambda3.  Sorry about that.